### ISSUE TITLE
Grouparray relies on an incorrect subquery ordering, and the sorting result is accurate only when the size is increased

### ISSUE THREAD

**Issue by liumaojing** [ClickHouse/ClickHouse#39400]
SELECT dip,　 sum(cnt2) AS cnt4,　 **groupArray(50)(dport) AS dport**,　        groupArray(50)(cnt2) AS cnt5,　        groupArray(50)(sip) AS sip,　        groupArray(50)(cnt3) AS cnt6　FROM　    (SELECT dip,　         dport,　        sum(cnt1) AS cnt2,　        groupArray(50)(sip) AS sip,　        groupArray(50)(cnt1) AS cnt3　    FROM　        (SELECT  dip,　        dport,　        sip,　         count() AS cnt1　        FROM t1　        WHERE ((`@timestamp` > '2022-07-14 00:30:04') AND (`@timestamp` <= '2022-07-15 00:30:04') )　        GROUP BY  dip,dport,sip　        ORDER BY  cnt1 DESC,sip ASC )　        GROUP BY  dip, dport　        ORDER BY  dport DESC)　    GROUP BY  dip　WITH TOTALS　ORDER BY  cnt4 DESC,dip DESC LIMIT 10 settings max_bytes_before_external_group_by=2073741824,max_bytes_before_external_sort=1073741824

the size is increased 500 ：groupArray(500)(dport)  result is accurate

version:21.3.14.1.7
The data accuracy ：above 1 billion

To sum up, grouparray size can affect the sorting result.
In addition,How does the groupingSets syntax currently set the default value to null and set size?

**Comment by zylthinking** [comment]
I found the following code, which means if the array is too huge to hold in memory, then parts of it will be written to temp file; then merged back via multi ISource --- which finally merged by groupArray.merge() together.

Seems order of array can be anything.
e.g.  groupArray can't keep the order in any case,  given the result array is large enough.

However, the document says:
In some cases, you can still rely on the order of execution. This applies to cases when SELECT comes from a subquery that uses ORDER BY.

```
        const auto & files = params->aggregator.getTemporaryFiles();
        Pipe pipe;

        {
            auto header = params->aggregator.getHeader(false);
            Pipes pipes;

            for (const auto & file : files.files)
                pipes.emplace_back(Pipe(std::make_unique<SourceFromNativeStream>(header, file->path())));

            pipe = Pipe::unitePipes(std::move(pipes));
        }
```

**Comment by alexey-milovidov** [comment]
Yes, the order is not guaranteed. I will update the docs.
